hail/src/main/scala/is/hail/services/DeployConfig.scala

```diff
@@ -0,0 +1,89 @@
+package is.hail.services
+
+import java.io.{File, FileInputStream}
+
+import is.hail.utils._
+import org.json4s.{DefaultFormats, Formats, JValue}
+import org.json4s.jackson.JsonMethods
+
+object DeployConfig {
+  lazy val get: DeployConfig = fromConfigFile()
+
+  def fromConfigFile(file0: String = null): DeployConfig = {
+    var file = file0
+
+    if (file == null)
+      file = System.getenv("HAIL_DEPLOY_CONFIG_FILE")
+
+    if (file == null) {
+      val fromHome = s"${ System.getenv("HOME") }/.hail/deploy-config.json"
+      if (new File(fromHome).exists())
+        file = fromHome
+    }
+
+    if (file != null) {
+      using(new FileInputStream(file)) { in =>
+        fromConfig(JsonMethods.parse(in))
+      }
+    } else
+      new DeployConfig(
+        "external",
+        "default",
+        Map())
+  }
+
+  def fromConfig(config: JValue): DeployConfig = {
+    implicit val formats: Formats = DefaultFormats
+    new DeployConfig(
+      (config \ "location").extract[String],
+      (config \ "default_namespace").extract[String],
+      (config \ "service_namespace").extract[Map[String, String]])
+  }
+}
+
+class DeployConfig(
+  val location: String,
+  val defaultNamespace: String,
```

Comment:
in the interest of consistency can we change the python `default_ns` to `default_namespace`?